pkg/controller/kubelet-config/kubelet_config_nodes.go

```diff
@@ -0,0 +1,336 @@
+
+		// updating the kubelet configuration with the Node specific configuration.
+		err = updateOriginalKubeConfigwithNodeConfig(nodeConfig, originalKubeConfig)
+		if err != nil {
+			return err
+		}
+
+		byteData, err := json.Marshal(originalKubeConfig)
+		if err != nil {
+			return err
+		}
+		kc.Spec.KubeletConfig.Raw = byteData
+		kc.ObjectMeta.Annotations = map[string]string{
+			ctrlcommon.GeneratedByControllerVersionAnnotationKey: version.Hash,
+		}
+		// Create or Update, on conflict retry
+		if err := retry.RetryOnConflict(updateBackoff, func() error {
+			var err error
+			if isNotFound {
+				_, err = ctrl.client.MachineconfigurationV1().KubeletConfigs().Create(context.TODO(), kc, metav1.CreateOptions{})
+			} else {
+				_, err = ctrl.client.MachineconfigurationV1().KubeletConfigs().Update(context.TODO(), kc, metav1.UpdateOptions{})
+			}
+			return err
+		}); err != nil {
+			return fmt.Errorf("Could not Create/Update KubeletConfig: %v", err)
+		}
+		glog.Infof("Applied Node configuration %v on MachineConfigPool %v", key, pool.Name)
+
+		// updating the nodeConfig status based on the latest kubelet config CR status
+		err = ctrl.updateNodeConfigStatus(kc, nodeConfig)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (ctrl *Controller) enqueueNodeConfig(nodeConfig *osev1.Node) {
+	key, err := cache.DeletionHandlingMetaNamespaceKeyFunc(nodeConfig)
+	if err != nil {
+		utilruntime.HandleError(fmt.Errorf("Couldn't get key for object %#v: %v", nodeConfig, err))
+		return
+	}
+	ctrl.nodeConfigQueue.Add(key)
+}
+
+func (ctrl *Controller) updateNodeConfig(old, cur interface{}) {
+	oldNode := old.(*osev1.Node)
+	newNode := cur.(*osev1.Node)
+	if !reflect.DeepEqual(oldNode.Spec, newNode.Spec) {
+		// skipping the update in case of the Worker-Latency-Profile type transition from "Default" to "LowUpdateSlowReaction" and vice-versa
+		// (TODO) Ideally the user request has to be honoured, the transition need to be from Default -> Medium -> Low or vice-versa.
+		// Restricting the request for now until this process is automated in future.
+		if (oldNode.Spec.WorkerLatencyProfile == osev1.DefaultUpdateDefaultReaction && newNode.Spec.WorkerLatencyProfile == osev1.LowUpdateSlowReaction) || (oldNode.Spec.WorkerLatencyProfile == osev1.LowUpdateSlowReaction && newNode.Spec.WorkerLatencyProfile == osev1.DefaultUpdateDefaultReaction) {
+			message := fmt.Sprintf("Skipping the Update Node event, name: %s, transition not allowed from old WorkerLatencyProfile: %s to new WorkerLatencyProfile: %s", newNode.Name, oldNode.Spec.WorkerLatencyProfile, newNode.Spec.WorkerLatencyProfile)
+			glog.Infof(message)
+			ctrl.updateNodeConfigDegradedStatus(newNode, message, "")
+			return
+		}
+		glog.V(4).Infof("Updating the node config resource, name: %s", newNode.Name)
+		ctrl.enqueueNodeConfig(newNode)
+	}
+}
+
+func (ctrl *Controller) addNodeConfig(obj interface{}) {
+	nodeConfig := obj.(*osev1.Node)
```

Comment:
We should be defensive here and use the `nodeConfig, ok` return to make sure it is a osev1.Node object.

Reply:
Sure, addressed

Reply:
Addressed